src/V3Tristate.cpp

```diff
@@ -486,8 +486,15 @@ class TristateVisitor final : public TristateBaseVisitor {
     AstVar* getCreateEnVarp(AstVar* invarp) {
         // Return the master __en for the specified input variable
         if (!invarp->user1p()) {
-            AstVar* const newp = new AstVar{invarp->fileline(), VVarType::MODULETEMP,
+            AstVar* const newp = new AstVar{invarp->fileline(),
+                                            v3Global.opt.pinsInoutEnables() ? VVarType::PORT
```

Comment:
I think you only want to do this at the top level module, not submodules.  Similar below.

Please add a hierarchical module to your test case that has tristates to test those aren't made into primary IO, thanks.

Reply:
I've added a sub-module to t_tri_top_en_out.v which doesn't propagate up to the top-level. This is tested in t_top_en_out_invalid.
I will have to go back an look into V3Tristate.cpp more as the __en outputs are incorrect.

Reply:
Done (internal_sub_io) checked in t_tri_top_en_out_invalid

Reply:
I still don't understand how this works, it is likely still making ports on inside modules.  (It might be nothing fails, but it's still wrong - look at some .tree files to make sure proper signals only are enables.)

Also try running some of the existing tristate tests with the new flags - they may fail because of .cpp misconnections, but should get through verilation ok.